client/engine/engine.go

```diff
@@ -128,9 +128,11 @@ func (e *Engine) ToApi() <-chan EngineEvent {
 func (e *Engine) Close() error {
 	close(e.stop)
 	close(e.toApi)
-	e.msg.Close()
-	e.chain.Close()
-	return nil
+	err := e.msg.Close()
+	if err != nil {
+		return err
+	}
+	return e.chain.Close()
 }
 
 // Run kicks of an infinite loop that waits for communications on the supplied channels, and handles them accordingly
```

Comment:
Nit: we could amend this godoc to describe the conditions under which the loop will exit.

Reply:
Good idea. Done in a99488d4